### ISSUE TITLE
PACKET_MMAP improvements

### ISSUE THREAD

**Issue by kevinGC** [google/gvisor#10966]
### Description

The fdbased LinkEndpoint supports PACKET_MMAP, but is missing a few useful improvements:

- TX: we TX using `sendmmsg`, but should use PACKET_MMAP buffers.
- RX: the PACKET_MMAP dispatcher seems to return 1-2 packets at a time, while the recvmmsg dispatcher often returns 4. We should investigate this, as PACKET_MMAP _should_ return many packets at once and ultimately be faster than recvmmsg.

### Is this feature related to a specific bug?

No

### Do you have a specific solution in mind?

See above

**Comment by kevinGC** [comment]
IIUC that's specific to V3. I am thinking that PACKET_MMAP improvements would be for the existing V2 interface we use.

**Comment by manninglucas** [comment]
Yeah could be. I definitely did less testing on V2 since I thought v3 would be the answer. FWIW when I tried to increase the number of frames (in v2) I didn't see any effect.

**Comment by hbhasker** [comment]
V3 is not going to work.v3 has a hard coded block time of 8ms? It's optimized for bulk packet capture (aka tcpdump). That's the reason I used v2. If you do a simple ping with V3 you will see the latency spike all over the place.

**Comment by kevinGC** [comment]
I've updated the title and description to make it clear that this issue is not related to use of V3, but rather is about improving the dispatcher using V1 or V2.
